tests/phpunit/tests/html-api/wpHtmlProcessor.php

```diff
@@ -0,0 +1,22 @@
+<?php
+/**
+ * Unit tests covering WP_HTML_Processor functionality.
+ *
+ * @package WordPress
+ * @subpackage HTML-API
+ */
+
+/**
+ * @group html-api
+ *
+ * @coversDefaultClass WP_HTML_Processor
+ */
+class Tests_HtmlApi_wpHtmlProcessor extends WP_UnitTestCase {
+	public function test_warns_not_to_call_the_public_constructor() {
```

Comment:
```suggestion
	/**
	 * Tests that calling the public constructor warns about calling the static creator methods.
	 *
	 * @ticket 58517
	 *
	 * @covers WP_HTML_Processor::__construct
	 */
	public function test_warns_not_to_call_the_public_constructor() {
```
Note: Using the `ClassName::method_name` format for the `@covers` annotation as this will help when restructuring the test suite.

Reply:
I might work this a bit differently, but thanks for the suggestion. In the rewrite here "warns about calling the static…" communicates to me the opposite intention, "don't call the static creator methods"

Probably "warns _to_ call the static creator methods" would resolve that.

Reply:
curious why we require the duplication here. if we want test names that describe what the test is doing, why also add it to the docblock? if we require the description in the docblock, why require the test name to match?

so far I've only been adding description to tests where there is something strange that the name alone doesn't explain; I also see that the vast majority of Core tests follow this same principle

Reply:
No doubt it's a good question. I took influence from the structure in [this ticket](https://core.trac.wordpress.org/ticket/56070).

Perhaps @SergeyBiryukov or @hellofromtonya could add their thoughts regarding this, if available?

Reply:
>I might work this a bit differently, but thanks for the suggestion. In the rewrite here "warns about calling the static…" communicates to me the opposite intention, "don't call the static creator methods"
> Probably "warns to call the static creator methods" would resolve that.

Ah, perhaps I misunderstood the `$message` parameter in the assertion. My bad!

Reply:
Updated everything in f8ee115376